File: src/mitris/sim/core/simulation/AbstractSimulator.java
package mitris.sim.core.simulation;

import mitris.sim.core.simulation.api.SimulatorInterface;
import mitris.sim.core.simulation.api.SimulationClock;

/**
 *
 * @author jlrisco
 */
public abstract class AbstractSimulator implements SimulatorInterface {

	protected SimulationClock clock;
	protected double tL; // Time of last event
	protected double tN; // Time of next event

	public AbstractSimulator(SimulationClock clock) {
		this.clock = clock;
	}

	public double getTL() {
		return tL;
	}

	public void setTL(double tL) {
		this.tL = tL;
	}

	public double getTN() {
		return tN;
	}

	public void setTN(double tN) {
		this.tN = tN;
	}

	public SimulationClock getClock() {
		return clock;
	}
}

File: simulation/abstract_simulator.go
package simulation

import "github.com/pointlesssoft/godevs/modeling"

type AbstractSimulator interface {
	Initialize()
	Exit()
	TA() float64
	Lambda()
	DeltFcn()
	Clear()
	GetModel() modeling.Component
	GetTL() float64
	SetTL(tl float64)
	GetTN() float64
	SetTN(tn float64)
	GetClock() Clock
}

func NewAbstractSimulator(clock Clock) AbstractSimulator {
	s := abstractSimulator{clock, 0, 0}
	return &s
}

type abstractSimulator struct {
	clock Clock
	tL float64  // Time of last event
	tN float64  // Time of next event
}

func (a *abstractSimulator) Initialize() {
	panic("implement me")
}

func (a *abstractSimulator) Exit() {
	panic("implement me")
}

func (a *abstractSimulator) TA() float64 {
	panic("implement me")
}

func (a *abstractSimulator) Lambda() {
	panic("implement me")
}

func (a *abstractSimulator) DeltFcn() {
	panic("implement me")
}

func (a *abstractSimulator) Clear() {
	panic("implement me")
}

func (a *abstractSimulator) GetModel() modeling.Component {
	panic("implement me")
}

func (a *abstractSimulator) GetTL() float64 {
	return a.tL
}

func (a *abstractSimulator) SetTL(tL float64) {
	a.tL = tL
}

func (a *abstractSimulator) GetTN() float64 {
	return a.tN
}

func (a *abstractSimulator) SetTN(tN float64) {
	a.tN = tN
}

func (a *abstractSimulator) GetClock() Clock {
	return a.clock
}
